Google.Cloud.Functions.Invoker/FunctionEnvironment.cs

```diff
@@ -61,6 +61,33 @@ internal sealed class FunctionEnvironment
         internal static FunctionEnvironment Create(Assembly functionAssembly, string[] commandLine, ConfigurationVariableProvider variableProvider) =>
             new Builder(functionAssembly, commandLine, variableProvider).Build();
 
+        /// <summary>
+        /// Attempts to find a single valid non-abstract function class within the given set of types.
+        /// (Note that "non-abstract class" is a pretty low bar; we could enforce non-generic etc, but we'll
```

Comment:
I agree this can wait but the error message they'll get might be confusing if the target type we find is generic for instance. It will fail when attempting to create an instance and some users might not even understand why an instance of that class is being created?

Reply:
Hmm. The error message isn't ideal, indeed:

> Unhandled exception. System.ArgumentException: Cannot create an instance of Google.Cloud.Functions.SampleHttpFunction.Function`1[T] because Type.ContainsGenericParameters is true.

But let's wait for a bit - I suspect we may want to change the instantiation model quite a bit in order to benefit from dependency injection, too... all of that could change things.